### PR TITLE
Add `metatensor.detach` operation

### LINKED ISSUE #149: `detach()` torch-based TensorMaps before saving them
When trying to save a TensorMap whose block values are `torch.Tensor` objects, an equistore error is raised if the values of the block have not been detached from their computational graph:

```py
from equistore import io
...
# tmp is a torch-based TensorMap
equistore.io.save("tmp.npz", tmp)
```
gives error:
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
File /opt/miniforge3/envs/rho/lib/python3.10/site-packages/equistore/utils.py:31, in catch_exceptions.<locals>.inner(*args, **kwargs)
     30 try:
---> 31     function(*args, **kwargs)
     32 except Exception as e:

File /opt/miniforge3/envs/rho/lib/python3.10/site-packages/equistore/data/array.py:138, in _eqs_array_data(this, data)
    134     # `.numpy()` will fail if the data is on GPU or requires gradient
    135     # tracking, and the resulting array is sharing data storage with the
    136     # tensor, meaning we can take a pointer to it without the array being
    137     # freed immediately.
--> 138     array = array.numpy()
    140 if not array.data.c_contiguous:

RuntimeError: Can't call numpy() on Tensor that requires grad. Use tensor.detach().numpy() instead.

The above exception was the direct cause of the following exception:

EquistoreError                            Traceback (most recent call last)
Cell In[23], line 6
      2 from rholearn import utils
      4 tmp = utils.tensor_to_torch(input, True, torch.float64, torch.device("cpu"))
----> 6 equistore.io.save("tmp.npz", tmp)

File /opt/miniforge3/envs/rho/lib/python3.10/site-packages/equistore/io.py:61, in save(path, tensor, use_numpy)
     59 else:
     60     lib = _get_library()
---> 61     lib.eqs_tensormap_save(path.encode("utf8"), tensor._ptr)

File /opt/miniforge3/envs/rho/lib/python3.10/site-packages/equistore/status.py:37, in _check_status(status)
     35 e = LAST_EXCEPTION
     36 LAST_EXCEPTION = None
---> 37 raise EquistoreError(last_error(), status) from e

EquistoreError: external error: calling eqs_array_t.data failed (status -1)
```

Ideally, as saving destroys pointers to the graph anyway, the `equistore.io.save` function should detach block values tensors before serialization, using `Tensor.detach()`.

### PR DESCRIPTION
**Pull request by Luthaf**
This also removed `requires_grad` from the `to` operation, preparing to move the operation to metatensor-torch and metatensor-core.

Fix #149.

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [x] CHANGELOG updated with public API or any other important changes?

----
:books: Documentation preview :books:: https://metatensor--424.org.readthedocs.build/en/424/

### CODE REVIEW

File: python/metatensor-operations/tests/detach.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        use_numpy=True,
+    )
+
+    # just checking that everything runs fine with numpy
+    tensor = metatensor.detach(tensor)
+
+    if HAS_TORCH:
+        tensor = metatensor.to(tensor, backend="torch")
+        for block in tensor:
+            block.values.requires_grad_(True)
```
Comment (jwa7) [inline python/metatensor-operations/tests/detach.py:30]: Could this be made into its own convenience operation, to provide opposite functionality to `detach()`?

File: docs/src/reference/operations/manipulation/index.rst [hunk 2]
```diff
@@ -8,6 +8,7 @@ Manipulation operations
     join() <join>
     manipulate dimension <manipulate-dimension>
     one_hot() <one-hot>
+    detach() <detach>
```
Comment (jwa7) [inline docs/src/reference/operations/manipulation/index.rst:11]: Alphabetical, before drop_blocks here?

**Review by jwa7 — CHANGES_REQUESTED** [review]

File: python/metatensor-operations/tests/detach.py (hunk 1 above)
Comment (Luthaf) [reply python/metatensor-operations/tests/detach.py]: done!

**Review by jwa7 — APPROVED** [review]
Looks good!

**Pull request merged**